src/java.desktop/share/classes/javax/imageio/stream/FileCacheImageOutputStream.java

```diff
@@ -217,6 +229,32 @@ public boolean isCachedMemory() {
         return false;
     }
 
+    private static class FileCacheDisposerRecord implements DisposerRecord {
+
+        private final File cacheFile;
+        private final RandomAccessFile cache;
+        private volatile boolean disposed;
+
+        public FileCacheDisposerRecord(File cacheFile, RandomAccessFile cache) {
```

Comment:
FileCacheDisposer makes more sense based on what the code does, but would StreamDisposerRecord be better to remain consistent for the input/output streams?